src/main/osd/osd.c

```diff
@@ -1205,23 +1209,23 @@ void osdUpdate(timeUs_t currentTimeUs)
 
         // Reset groupings
         for (elementGroup = 0; elementGroup < OSD_GROUP_COUNT; elementGroup++) {
-            if (osdElementGroupDurationUs[elementGroup] > (osdElementGroupTargetUs[elementGroup] + OSD_ELEMENT_RENDER_GROUP_MARGIN)) {
-                osdElementGroupDurationUs[elementGroup] = 0;
+            if (osdElementGroupDurationFracUs[elementGroup] > (OSD_ELEMENT_RENDER_TARGET << OSD_EXEC_TIME_SHIFT)) {
```

Comment:
(OSD_ELEMENT_RENDER_TARGET << OSD_EXEC_TIME_SHIFT)
So we reset at (30 << 8) == 7680μs?

Reply:
Not quite; this code maintains fractional us values so we can reduce the task estimates very slowly after they peak (for whatever reason). For the OSD task we want this decay to be very slow as the tasks are called as a high rate, so we maintain the estimates in `1 /(1 << 8)` us. 7680 divided by that is 30 us.